test/unit/distributed_tracing/dt-cats.test.js

```diff
@@ -64,7 +68,10 @@ describe('distributed tracing', function() {
 
         if (testCase.outbound_payloads) {
           testCase.outbound_payloads.forEach((outbound) => {
-            const created = JSON.parse(tx.createDistributedTracePayload().text())
+            let headers = {}
+            tx.insertDistributedTraceHeaders(headers)
+            const payload = headers.newrelic
+            const created = tx._getParsedPayload(payload)
```

Comment:
I update these cross agent tests to use the "headers" methods. One effect of this was that the `acceptDistributedTraceHeaders()` technically had a bug. If the `newrelic` header was present, but was an empty string, the agent should send up a `DistributedTrace/AcceptPayload/Ignored/Null` metric. But the way the if statement was setup, if the newrelic "payload" was empty it would skip processing all together.

Reply:
Nice catch and update!